test/e2e/specs/editor/various/preview.spec.js

```diff
@@ -0,0 +1,342 @@
+
+		await previewPage.close();
+	} );
+
+	test( 'should not revert title during a preview right after a save draft', async ( {
+		editor,
+		page,
+		pageUtils,
+		previewUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Type aaaaa in the title field.
+		await editorPage.type( 'role=textbox[name="Add title"]', 'aaaaa' );
+		await editorPage.keyboard.press( 'Tab' );
+
+		// Save the post as a draft.
+		await editorPage.click( 'role=button[name="Save draft"i]' );
+		await editorPage.waitForSelector(
+			'role=button[name="Dismiss this notice"] >> text=Draft saved'
+		);
+
+		// Open the preview page.
+		const previewPage = await editor.openPreviewPage( editorPage );
+
+		// Title in preview should match input.
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		await expect( previewTitle ).toHaveText( 'aaaaa' );
+
+		// Return to editor.
+		await editorPage.bringToFront();
+
+		// Append bbbbb to the title, and tab away from the title so blur event is triggered.
+		await editorPage.focus( 'role=textbox[name="Add title"]' );
+		await pageUtils.pressKeyWithModifier( 'primary', 'a' );
+		await editorPage.keyboard.press( 'ArrowRight' );
+		await editorPage.keyboard.type( 'bbbbb' );
+		await editorPage.keyboard.press( 'Tab' );
+
+		// Save draft and open the preview page right after.
+		await editorPage.click( 'role=button[name="Save draft"i]' );
+		await editorPage.waitForSelector(
+			'role=button[name="Dismiss this notice"] >> text=Draft saved'
+		);
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		await expect( previewTitle ).toHaveText( 'aaaaabbbbb' );
+
+		await previewPage.close();
+	} );
+
+	// Verify correct preview. See: https://github.com/WordPress/gutenberg/issues/33616
+	test( 'should display the correct preview when switching between published and draft statuses', async ( {
+		editor,
+		page,
+		previewUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Type Lorem in the title field.
+		await editorPage.type( 'role=textbox[name="Add title"]', 'Lorem' );
+
+		// Open the preview page.
+		const previewPage = await editor.openPreviewPage( editorPage );
+
+		// Title in preview should match input.
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		await expect( previewTitle ).toHaveText( 'Lorem' );
+
+		// Return to editor and publish post.
+		await editorPage.bringToFront();
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.click( 'role=button[name="Close panel"i]' );
+
+		// Change the title and preview again.
+		await editorPage.type( '[aria-label="Add title"]', ' Ipsum' );
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match updated input.
+		await expect( previewTitle ).toHaveText( 'Lorem Ipsum' );
+
+		// Return to editor and switch to Draft.
+		await editorPage.bringToFront();
+		await page.click( 'role=button[name="Switch to draft"i]' );
```

Comment:
Seems like we missed this:

```suggestion
		await page.click( 'role=button[name="Switch to draft"i]' );
		await page.click( 'role=dialog >> role=button[name="OK"i]' );
```

Reply:
Oooh I see, thanks!